include/ginkgo/core/base/executor.hpp

```diff
@@ -797,6 +797,18 @@ class Executor : public log::EnableLogging<Executor> {
         return this->verify_memory_from(other.get());
     }
 
+    /** Sets the logger propagation mode for this executor. */
+    void set_log_propagate_mode(log::propagate_mode mode)
```

Comment:
Yes, I would prefer that. I think it doesn't match with our logging philosophy.

Reply:
Do you see a technical solution to doing this? Should profiling loggers be their own category with their own add/remove functions? I'm not sure I would like that, even though it is a simple solution.

Reply:
I was thinking of checking in the `add_logger` for executors if the passed-in logger needs events propagated. The check could be just if its runtime-type is `ProfilerHook`. (And also in the `remove_logger`)

Reply:
That means adding a virtual function `bool needs_propagation()` to the logger?

Reply:
Right now `if(auto p = dynamic_cast<ProfilerHook>(logger))` would be enough.

Reply:
yes, but I would prefer a more extensible approach ;)

Reply:
We could also add an override `add_logger(std::shared_ptr<ProfilerHook>)`. But I would be fine with any of these approaches.

Reply:
I somehow disagree propagate all logger to executor, especially on the convergence.
When there are several solver (or solver with the preconditioner), it will confusing at which LinOp convergence is logged by the solver.